src/main/java/com/mg105/entities/Battle.java

```diff
@@ -0,0 +1,71 @@
+package com.mg105.entities;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class Battle {
+    //battlestatus: 0 for in progress, -1 for loss, 1 for win
+    private int battleStatus;
+    private final ArrayList<BattleCharacter> opponents;
+    private final ArrayList<BattleCharacter> playerCharacters;
+    private final ArrayList<BattleCharacter> moveQueue = new ArrayList<BattleCharacter>();
+
+    //not sure if numTokens should be randomly generated on creation, or after battleStatus has changed (this
+    // implementation will assume the latter).
+    private int numTokens;
+
+    public Battle(ArrayList<BattleCharacter> opponents, ArrayList<BattleCharacter> party) {
+        this.battleStatus = 0;
+        this.opponents = opponents;
+        this.playerCharacters = party;
+        this.numTokens = 0;
+
+        this.moveQueue.addAll(this.opponents);
+        this.moveQueue.addAll(playerCharacters);
+        Collections.sort(moveQueue);
+    }
+
+    public int getBattleStatus() {
+        return battleStatus;
+    }
```

Comment:
Personally, I prefer writing an if else statement and return "win", "lose" or "in progress" in string form but it is fine for returning -1,0,1 coercion.